app/src/main/java/io/gnosis/safe/ui/safe/overview/SafeOverviewViewModel.kt

```diff
@@ -0,0 +1,39 @@
+package io.gnosis.safe.ui.safe.overview
+
+import androidx.lifecycle.ViewModel
+import pm.gnosis.svalinn.common.PreferencesManager
+import pm.gnosis.svalinn.common.utils.edit
+import javax.inject.Inject
+
+class SafeOverviewViewModel
+@Inject
+constructor(
+    private val preferencesManager: PreferencesManager
+) : ViewModel() {
+
+    lateinit var termsBottomSheetDialog: TermsBottomSheetDialog
```

Comment:
Let's not keep reference to views in the viewModels. This can lead to leaks and is a bit of an anti pattern under MVVM. Then for testing, you will have through 7 different types of mocking hell to verify something that you could just do by emitting a data class instance as a view state update. I can show you around the app if you want an example.